### PR TITLE
Glide with pinned dependencies.

### PR DESCRIPTION
**Pull request by nanliu**
Replace godep with glide for dependency management.

* unlike godep restore, glide will not change dependency repository in go path.
* glide will detect dependency conflicts.
* glide provides a path forward for vendoring options.

Testing done:
- unit
- manual

@intelsdi-x/snap-maintainers

### CODE REVIEW

**Comment by nanliu** [comment]
please test this.

**Comment by nanliu** [comment]
This is ready for review again. The commit history is left because we were exploring using floating versions and decided it's too much change in one step and introduced new issues. I can squash it when we decide it's ready to merge.

**Comment by geauxvirtual** [comment]
LGTM. I've tested this on Mac OS X and Linux.

**Comment by nanliu** [comment]
Pending discussion during weekly sync with Poland team.

**Comment by nanliu** [comment]
Reran the travis.ci test and it does not appear to need any updates prior to merging.

File: scripts/deps.sh [hunk 1]
```diff
@@ -29,7 +29,7 @@ __proj_dir="$(dirname "$__dir")"
 
 _go_path
 
-_go_get github.com/tools/godep
+_go_get github.com/Masterminds/glide
 
-_info "restoring dependency with godep"
-(cd "${__proj_dir}" && godep restore)
+_info "restoring dependency with glide"
+(cd "${__proj_dir}" && glide install)
```
Comment (candysmurf) [inline scripts/deps.sh]: Do you need glide update here too in case version info manually changed?

**Comment by candysmurf** [comment]
👍 , LGTM.  small, medium and legacy tests all passed in my Mac.

**Comment by andrzej-k** [comment]
LGTM

**Pull request merged**